src/core/thread/child_table.cpp

```diff
@@ -102,7 +102,11 @@ ChildTable::ChildTable(Instance &aInstance)
     : InstanceLocator(aInstance)
     , mMaxChildrenAllowed(kMaxChildren)
 {
-    Clear();
+    for (Child *child = &mChildren[0]; child < OT_ARRAY_END(mChildren); child++)
+    {
+        child->Init(aInstance);
+        child->Clear();
```

Comment:
Does this actually work when `OPENTHREAD_CONFIG_MULTIPLE_INSTANCE_ENABLE=1`? The current implementation of `Child::Clear()` simply calls `memset()`.
https://github.com/openthread/openthread/blob/0823114dfff39dd7d27eeaf1ba4ac3d16002ba86/src/core/thread/topology.cpp#L106-L110